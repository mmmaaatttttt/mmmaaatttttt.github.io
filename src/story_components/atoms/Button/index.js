import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import { darken } from "polished";

const color = props => props.color;
const darkerColor = props => darken(0.1, color(props));

const Button = styled.button`
  border-radius: 3px;
  box-sizing: border-box;
  padding: 0.2rem 0.6rem;
  background-color: ${color};
  border: 1px solid ${color};
  color: white;
  font-size: 80%;
  line-height: 1.4rem;
  margin: 0.5rem;

  &:active,
  &:focus {
    outline: none;
  }

  &:hover {
    cursor: pointer;
    background-color: ${darkerColor};
  }

  ${props =>
    props.large &&
    css`
      width: 100%;
      padding: 0.5rem 0.6rem;
      margin: 0.5rem 0;
    `}

  ${props =>
    props.color === "white" &&
    css`
      color: black;
      border: 1px solid black;
    `} ${props =>
  props.disabled &&
  css`
    opacity: 0.7;

    &:hover {
      cursor: not-allowed;
    }
  `};
`;

Button.propTypes = {
  color: PropTypes.string.isRequired,
  large: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired
};

Button.defaultProps = {
  color: "white",
  large: false,
  disabled: false
};

export default Button;
