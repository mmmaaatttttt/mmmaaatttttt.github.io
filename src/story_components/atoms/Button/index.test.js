import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from ".";

it("renders successfully", () => {
  shallow(<Button />);
});

it("matches snapshot with props passed in", () => {
  const wrapper = shallow(
    <Button color="blue" large disabled>
      I'm a button
    </Button>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
