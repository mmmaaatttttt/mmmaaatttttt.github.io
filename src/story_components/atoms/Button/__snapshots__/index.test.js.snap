// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props passed in 1`] = `
<StyledComponent
  color="blue"
  disabled={true}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  border-radius: 3px;
  box-sizing: border-box;
  padding: 0.2rem 0.6rem;
  background-color: ",
          [Function],
          ";
  border: 1px solid ",
          [Function],
          ";
  color: white;
  font-size: 80%;
  line-height: 1.4rem;
  margin: 0.5rem;

  &:active,
  &:focus {
    outline: none;
  }

  &:hover {
    cursor: pointer;
    background-color: ",
          [Function],
          ";
  }

  ",
          [Function],
          "

  ",
          [Function],
          " ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "color": "white",
        "disabled": false,
        "large": false,
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "color": [Function],
        "disabled": [Function],
        "large": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  large={true}
>
  I'm a button
</StyledComponent>
`;
